torchmetrics/text/rouge.py

```diff
@@ -99,31 +110,45 @@ def __init__(
             if key not in ALLOWED_ROUGE_KEYS:
                 raise ValueError(f"Got unknown rouge key {key}. Expected to be one of {ALLOWED_ROUGE_KEYS}")
 
+        if accumulate not in ALLOWED_ACCUMULATE_VALUES:
+            raise ValueError(
+                f"Got unknown accumulate value {accumulate}. Expected to be on of {ALLOWED_ACCUMULATE_VALUES}"
+            )
+
         self.rouge_keys = rouge_keys
         self.rouge_keys_values = [ALLOWED_ROUGE_KEYS[key] for key in rouge_keys]
         self.stemmer = nltk.stem.porter.PorterStemmer() if use_stemmer else None
+        self.accumulate = accumulate
 
         # Adding stated dynamically to prevent IndexError during sync function as some lists can be empty.
         for rouge_key in self.rouge_keys:
             for score in ["fmeasure", "precision", "recall"]:
                 self.add_state(f"{rouge_key}_{score}", [], dist_reduce_fx=None)
 
-    def update(self, preds: Union[str, List[str]], targets: Union[str, List[str]]) -> None:  # type: ignore
+    def update(
+        self, preds: Union[str, List[str]], targets: Union[str, List[str], List[List[str]]]
+    ) -> None:  # type: ignore
         """Compute rouge scores.
 
         Args:
             preds: An iterable of predicted sentences or a single predicted sentence.
             targets: An iterable of target sentences or a single target sentence.
         """
 
+        if isinstance(targets, list) and bool(targets) and all(isinstance(elem, str) for elem in targets):
+            if isinstance(preds, str):
+                targets = [str(x) for x in targets]
+            else:
+                targets = [[str(x)] for x in targets]
+
```

Comment:
```suggestion
    if isinstance(targets, list) and all(isinstance(target, str) for target in targets) and isinstance(preds, list):
        targets = [[target] for target in targets]

```

We shouldn't be required to call `str(target)` as this condition is already verified by the preceding condition.